core/templates/dev/head/domain/skill/ConceptCardBackendApiService.ts

```diff
@@ -52,65 +52,97 @@ oppia.factory('ConceptCardBackendApiService', [
       return _conceptCardCache.hasOwnProperty(skillId);
     };
 
+    var _getUncachedSkillIds = function(skillIds) {
+      var uncachedSkillIds = [];
+      skillIds.forEach(function(skillId) {
+        if (!_isCached(skillId)) {
+          uncachedSkillIds.push(skillId);
+        }
+      });
+      return uncachedSkillIds;
+    };
+
     return {
       /**
-       * Retrieves the concept card of a skill from the backend given a skill
-       * ID. This returns a promise object that allows a success and rejection
-       * callbacks to be registered. If the concept card is successfully loaded
-       * and a success callback function is provided to the promise object, the
-       * success callback is called with the concept card passed in as a
-       * parameter. If something goes wrong while trying to fetch the
-       * concept card, the rejection callback is called instead, if present. The
-       * rejection callback function is passed the error that occurred and the
-       * skill ID.
+       * Retrieves the concept cards of skills from the backend given the skill
+       * IDs. This returns a promise object that allows a success and rejection
+       * callbacks to be registered. If the concept cards are successfully
+       * loaded and a success callback function is provided to the promise
+       * object, the success callback is called with the concept card passed
+       * in as a parameter. If something goes wrong while trying to fetch the
+       * concept cards, the rejection callback is called instead, if present.
+       * The rejection callback function is passed the error that occurred and
+       * the skill IDs.
        */
-      fetchConceptCard: function(skillId) {
+      fetchConceptCards: function(skillIds) {
```

Comment:
Is this really needed? Since loadConceptCards would fetch it anyway if all skill ids given aren't cached.

Reply:
Removed this bit and replaced `fetchedConceptCards` with `loadConceptCards` throughout the codebase.